### PR TITLE
[DO NOT MERGE] Docs for server 4.1

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
This is the docs release branch for server v4.1.0

All docs are in `_cci2/server/4.1/`

Please branch off this branch for any changes for this release.

New pages:

* Air gapped installation guide directory

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/server/v4.1/installation/phase-1-prerequisites.adoc [hunk 1]
```diff
@@ -0,0 +1,889 @@
...
+gcloud iam service-accounts keys create <KEY_FILE> \
+    --iam-account circleci-storage@<PROJECT_ID>.iam.gserviceaccount.com
+----
+--
+endif::env-aws[]
+
+ifndef::pdf[]
+[#next-steps]
+== Next steps
+* xref:server/v4.1/installation/phase-2-core-services#[Phase 2: Core Services Installation].
```
Comment (rosieyohannan) [inline jekyll/_cci2/server/v4.1/installation/phase-1-prerequisites.adoc:888]: @rsylvian here's an example of a cross reference. This works fine to get to: `circleci.com/docs/server/v4.1/installation/phase-2-core-services/` but fails the html proofer

File: jekyll/_data/sidenav.yml [hunk 2]
```diff
@@ -566,16 +566,94 @@ en:
...
 
-  - name: Server administration v4.x
+  - name: Server administration v4.1
     icon: icons/sidebar/admin-new.svg
     children:
-      - name: Server v4.x overview
+      - name: Server v4.1 overview
+        children:
+          - name: CircleCI server v4.x overview
+            link: server/v4.1/overview/circleci-server-v4-overview
```
Comment (kbatuigas) [inline jekyll/_data/sidenav.yml:575]: The filename doesn't have v4

**Review by kbatuigas — CHANGES_REQUESTED** [review]

File: jekyll/_cci2/server/v4.1/air-gapped-installation/phase-3-install-circleci-server.adoc [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+* Set `container.registry` to the internal registry hostname (and port if using a non-standard port) of your instance
+
+[source, yaml]
+----
+global:
+  ...
+  domainName: "server.internal.example.com"
+  container:
+    registry: "docker.internal.example.com:5000"
+    org: "circleci"
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/air-gapped-installation/phase-3-install-circleci-server.adoc:38]:
```suggestion
    org: "<image-registry-org>"
```

File: jekyll/_cci2/server/v4.1/air-gapped-installation/example-values.yaml.adoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+[source, yaml]
+----
+
+# Private docker registry at docker.internal.example.com:5000
+global:
+  domainName: "server.internal.example.com"
+  license: '<<your-server-license-here>>'
+  container:
+    registry: "docker.internal.example.com:5000"
+    org: "circleci"
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/air-gapped-installation/example-values.yaml.adoc:42]:
```suggestion
    org: "<image-registry-org>"
```

File: jekyll/_cci2/server/v4.1/air-gapped-installation/phase-2-configure-object-storage.adoc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+wget https://circleci-binary-releases.s3.amazonaws.com/circleci-agent/$LATEST_RELEASE/checksums.txt
+----
+
+[#create-release-dir]
+=== d. Create a release directory in the circleci-data bucket
+Using the version specified in `release.txt`, create a new directory in the root of the `circleci-data` bucket with the name of that release.
+
+```
+# In this example, we are creating a directory at the root of circleci-data in MinIO with the following name:
+1.0.137184-db08738f
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.1/air-gapped-installation/phase-2-configure-object-storage.adoc:73]: I've moved the contents of this code block into the regular text, but yes `1.0.137184-db08738f` doesn't match what's in the screenshot. I imagine this is just incorrect/left over from a previous iteration. @circlery should these numbers match? I will update the one in the text to match the screenshot if so :-D

File: jekyll/_cci2/server/v4.1/air-gapped-installation/phase-2-configure-object-storage.adoc [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+image::./minio/minio_create_release_dir.png[Creating a directory in the circleci-data bucket]
+
+[#upload-checksums-file]
+=== e. Upload the checksums.txt file to the newly created directory.
+Copy the downloaded `checksums.txt` file (step c) to your virtual environment, and place it in MinIO nested under the newly created release directory (step d).
+
+```
+# The structure of the directory should now look like this
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.1/air-gapped-installation/phase-2-configure-object-storage.adoc:83]: Yeah that will work, I'll update 👍

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+## Helm Diff (optional)
+The Helm Diff tool is used to verify that the changes between your current install and the upgrade are expected.
+
+# diff command
+helm diff upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml --show-secrets --context 5 oci://cciserver.azurecr.io/circleci-server --version 4.0.1
+-------------------------------------------------------------------------
+## Helm Upgrade CircleCI Server
+helm upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml oci://cciserver.azurecr.io/circleci-server --version <version-to-upgrade-to> --force
+
+NOTE: After server 3.x to 4.x migration, You must rerun the Nomad terraform with modified value of 'server_endpoint' variable
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:101]: ooo good spot, I imagine so!

File: jekyll/_data/sidenav.yml [hunk 8]
```diff
@@ -566,16 +566,94 @@ en:
...
+          - name: Release notes
+            link: server/v4.1/overview/release-notes
+      - name: Server v4.1 installation
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_data/sidenav.yml:578]: Oh yeah good spot! Must have just been missed off. I'll add in for 4.1 and fix 4.0 separately

File: jekyll/_cci2/server/v4.1/air-gapped-installation/phase-2-configure-object-storage.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (circlery) [reply jekyll/_cci2/server/v4.1/air-gapped-installation/phase-2-configure-object-storage.adoc]: The numbers should match whatever is the latest in release.txt which is currently `1.0.164633-f867f14f`. 

It's in a code block to highlight the hierarchy of the file structure - but if there's a better way to do this please feel free to modify it.

File: jekyll/_cci2/server/v4.1/air-gapped-installation/additional-considerations.adoc [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+----
+{
+      "insecure-registries":["docker.example.internal:5000"]
+}
+----
+
+This file will need to be configured on the following machines:
+
+- All Nomad nodes in the air-gapped environment
+- Potentially all K3s nodes in the air-gapped environment, if using docker-backed K3s
```
[the remark this answers is not in the document]
Comment (circlery) [reply jekyll/_cci2/server/v4.1/air-gapped-installation/additional-considerations.adoc:34]: @atulsingh0 This won't be possible without modifying several other sections downstream. This airgapped tutorial uses K3s as a kubernetes provider, and it's unfortunately not as simple as removing the reference.

**Review by circlery — APPROVED** [review]

File: jekyll/_cci2/server/v4.1/overview/circleci-server-overview.adoc [hunk 10]
```diff
@@ -0,0 +1,389 @@
+---
+contentTags:
+  platform:
+    - Server v4.1
+    - Server Admin
+---
+= CircleCI server v{serverversion41} overview
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.1/overview/circleci-server-overview.adoc:7]: One last thing - not sure how this works but this shows up wonky for me

image

File: jekyll/_cci2/server/v4.1/overview/circleci-server-overview.adoc [hunk 11]
```diff
@@ -0,0 +1,389 @@
...
+| Should work with Minio and Runner.
+|===
+
+ifndef::pdf[]
+
+[#next-steps]
+== Next steps
+
+* xref:release-notes#[CircleCI server v4.1 release notes]
+* xref:../installation/phase-1-prerequisites#[Server 4.x Installation Prerequisites]
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/overview/circleci-server-overview.adoc:387]:
```suggestion
* xref:../installation/phase-1-prerequisites#[Server 4.1 Installation Prerequisites]
```

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+The Helm Diff tool is used to verify that the changes between your current install and the upgrade are expected.
+
+# diff command
+helm diff upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml --show-secrets --context 5 oci://cciserver.azurecr.io/circleci-server --version 4.0.1
+-------------------------------------------------------------------------
+## Helm Upgrade CircleCI Server
+helm upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml oci://cciserver.azurecr.io/circleci-server --version <version-to-upgrade-to> --force
+----
+
+NOTE: After server 3.x to 4.x migration, You must rerun the Nomad terraform with modified value of 'server_endpoint' variable
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:102]:
```suggestion
NOTE: After server 3.x to 4.x migration, you must rerun the Nomad terraform with the modified value of the `server_endpoint` variable. It should be - `<domain-name>:4647`.
```
I actually missed this commit, sorry

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+It should be - <domain-name>:4647
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:103]:
```suggestion

```

**Review by kbatuigas — APPROVED** [review]
Just one nit, but approved 🎉

File: jekyll/_cci2/server/v4.1/operator/operator-overview.adoc [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+:page-liquid:
+:page-description: Learn about the various tasks and tools involved in administering an installation of CircleCI server.
+:icons: font
+:toc: macro
+:toc-title:
+
+This guide contains information for CircleCI server operators, or those responsible for ensuring CircleCI server is running properly through maintenance and monitoring. Before reading this operator guide, ensure you have read the xref:../overview/circleci-server-overview#[CircleCI server v4.1 overview].
+
+CircleCI server schedules CI/CD jobs using the https://www.nomadproject.io/[Nomad] scheduler. The Nomad Server (control plane) runs inside the Kubernetes cluster, while Nomad clients are provisioned outside the cluster. The Nomad clients need access to the Nomad control plane, output processor,
+and VM service.
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/operator/operator-overview.adoc:18]:
```suggestion
CircleCI server schedules CI/CD jobs using the https://www.nomadproject.io/[Nomad] scheduler. The Nomad Server (control plane) runs inside the Kubernetes cluster, while Nomad clients are provisioned outside the cluster. The Nomad clients need access to the Nomad control plane, output processor, and VM service.
```

File: jekyll/_cci2/server/v4.1/operator/introduction-to-nomad-cluster-operation.adoc [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+- `dead`: Nomad has finished executing the job. The status becomes `dead` regardless of whether the corresponding CircleCI job/build succeeds or fails.
+
+[#checking-the-cluster-status]
+=== Checking the cluster status
+
+To get a list of your Nomad clients, run the following command:
+
+[source,shell]
+----
+kubectl exec -it nomad-server-pod-ID -- nomad node-status
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/operator/introduction-to-nomad-cluster-operation.adoc:63]:
```suggestion
kubectl exec -it <<nomad-server-pod-ID>> -- nomad node-status
```

**Review by atulsingh0 — APPROVED** [review]

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+- <path>/kots-exporter/output/helm-values.yaml
+
+## Helm login to cciserver.azurecr.io
+helm registry login cciserver.azurecr.io --username <image-registry-username>
+
+## Helm Diff (optional)
+The Helm Diff tool is used to verify that the changes between your current install and the upgrade are expected.
+
+# diff command
+helm diff upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml --show-secrets --context 5 oci://cciserver.azurecr.io/circleci-server --version 4.0.1
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:96]:
```suggestion
helm diff upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml --show-secrets --context 5 oci://cciserver.azurecr.io/circleci-server --version 4.0.2
```

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+-------------------------------------------------------------------------
+## Helm Upgrade CircleCI Server
+helm upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml oci://cciserver.azurecr.io/circleci-server --version <version-to-upgrade-to> --force
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:99]:
```suggestion
helm upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml oci://cciserver.azurecr.io/circleci-server --version 4.0.2 --force
```

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 18]
```diff
@@ -0,0 +1,235 @@
...
+
+[#generate-helm-diff-output]
+=== 5. Generate helm-diff output
+Next, execute the helm-diff command and review the output.
+
+[source,shell]
+----
+helm registry login cciserver.azurecr.io -u <image-registry-username>
+
+helm diff upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml --show-secrets --context 5 oci://cciserver.azurecr.io/circleci-server --version 4.0.1
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:145]:
```suggestion
helm diff upgrade <release-name> -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml --show-secrets --context 5 oci://cciserver.azurecr.io/circleci-server --version 4.0.2
```

File: jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc [hunk 19]
```diff
@@ -0,0 +1,235 @@
...
+** Redis (redis-master, redis-slave)
+
+[#upgrading-circleci-server-3]
+=== 6. Upgrading CircleCI server 3.x
+Once your helm-value file is verified, run the following commands to upgrade the CircleCI server to v4.x.
+
+Our helm registry is stored in an azure private registry. You will be provided a username and token to access the registry.
+[source,shell]
+----
+helm upgrade circleci-server -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml oci://cciserver.azurecr.io/circleci-server --version 4.0.1 --force
```
Comment (atulsingh0) [inline jekyll/_cci2/server/v4.1/installation/migrate-from-server-3-to-server-4.adoc:179]:
```suggestion
helm upgrade circleci-server -n <namespace> -f <path>/kots-exporter/output/helm-values.yaml oci://cciserver.azurecr.io/circleci-server --version 4.0.2 --force
```

**Pull request merged**